suprb/optimizer/rule/ns/ns.py

```diff
@@ -122,13 +99,18 @@ def optimize(self, X: np.ndarray, y: np.ndarray, n_rules: int) -> list[Rule]:
             Return: A list of filtered Rules.
         """
         self._validate_params(n_rules)
-
         self.random_state_ = check_random_state(self.random_state)
+        self.novelty_calculation.archive.set_random_state(self.random_state_)
+        self.novelty_calculation.archive.set_archive(self.pool_)
```

Comment:
should we use a copy of pool here? we are changing the objects within the archive (at least by adding a distance parameter)

Reply:
As mentioned above, we are using a copy of the archive in `_novelty_score`, so we don't alter the pool. But I added `deepcopy` so the pool doesn't get accidentally changed